packages/components/src/internal/components/user/ProfilePage.tsx

```diff
@@ -30,6 +35,289 @@ import { UserProfile } from './UserProfile';
 import { ChangePasswordModal } from './ChangePasswordModal';
 
 import { useUserProperties } from './hooks';
+import { Alert } from '../base/Alert';
+import { ActionURL } from '@labkey/api';
+import { AppContext, useAppContext } from '../../AppContext';
+import { setCopyValue } from '../../events';
+import { getPrimaryAppProperties, isFeatureEnabled } from '../../app/utils';
+import { ProductFeature } from '../../app/constants';
+import {
+    InjectedQueryModels,
+    QueryConfigMap,
+    RequiresModelAndActions,
+    withQueryModels
+} from '../../../public/QueryModel/withQueryModels';
+import { SCHEMAS } from '../../schemas';
+import { GridPanel } from '../../../public/QueryModel/GridPanel';
+import { ConfirmModal } from '../base/ConfirmModal';
+import { HelpLink } from '../../util/helpLinks';
+
+interface ButtonsComponentProps extends RequiresModelAndActions {
+    onDelete: () => void;
+}
+
+const APIKeysButtonsComponent: FC<ButtonsComponentProps> = props => {
+    const { model, actions, onDelete } = props;
+    const { api } = useAppContext<AppContext>();
+    const [ showConfirmDelete, setShowConfirmDelete ] = useState<boolean>(false);
+
+    const onDeleteClicked = useCallback(() => setShowConfirmDelete(true), []);
+    const closeDeleteModal = useCallback(() => setShowConfirmDelete(false), []);
+    const onConfirmDelete = useCallback(async () => {
+        await api.security.deleteApiKeys(model.selections);
+        actions.clearSelections(model.id);
+        actions.loadModel(model.id, true, true);
+        actions.loadFirstPage(model.id);
```

Comment:
is this `loadFirstpage` needed after the call to `loadModel`? Just curious as I don't recall doing this for other delete actions (but maybe we should be).

Reply:
It seemed to be required if I'm on the second page and delete the one remaining entry on that page. It was staying on that page and saying there was no data. I'll try again to be sure.